src/shared/components/MolecularProfileSelector.tsx

```diff
@@ -0,0 +1,75 @@
+import {observer} from "mobx-react";
+import * as React from "react";
+import {MolecularProfile} from "../api/generated/CBioPortalAPI";
+import {MobxPromise} from "mobxpromise";
+import {remoteData} from "../api/remoteData";
+import Select from "react-select";
+import LoadingIndicator from "./loadingIndicator/LoadingIndicator";
+
+
+export interface IMolecularProfileSelector {
+    name?:string;
+    className?:string;
+    value:string;
+    onChange:(option:{label:string, value:string})=>void;
+    molecularProfiles:MolecularProfile[] | MobxPromise<MolecularProfile[]>;
+    molecularProfileIdToProfiledSampleCount:{[molecularProfileId:string]:number} | MobxPromise<{[molecularProfileId:string]:number}>;
+}
+
+@observer
+export default class MolecularProfileSelector extends React.Component<IMolecularProfileSelector, {}> {
+    readonly resolvedPromise = remoteData(()=>Promise.resolve());
+
+    readonly options = remoteData({
```

Comment:
why not just deal with async loading of options outside of component?

Reply:
the main functionality here is to let anyone use the same component when dealing w molecular profiles and available sample data counts. we want to encapsulate that logic